### ISSUE TITLE
"You are not allowed to change your password. Please contact your system administrator."

### ISSUE THREAD

**Issue by keagylabs** [unosquare/passcore#594]
**PassCore Server**
- OS: Windows Server 2019 Standard Edition
- Provider: Active Directory
- Settings file (without sensitive information):
{
  "Logging": {
    "IncludeScopes": false,
    "LogLevel": {
      "Default": "Debug",
      "System": "Information",
      "Microsoft": "Information"
    }
  },
  "WebSettings": {
    "EnableHttpsRedirect": true
  },
  "AppSettings": {
    // The following options for AD Provider (remove if you don't use this Provider)
    "UseAutomaticContext": true, // Set true to allow PassCore to reset password using the same credentials, or false if you will fill the credentials below
    "RestrictedADGroups": [
      "Administrators",
      "Domain Admins",
      "Enterprise Admins"
    ], // Set the AD groups to restrict the use of PassCore
    "AllowedADGroups": [], // Set the AD Groups to allow PassCore, if the array is empty all the groups no-restricted above are allowed
    "IdTypeForUser": "UPN", // Possible values are "DN", "GUID", "Name", "SAM", "SID" and "UPN" (Default UPN)
    "UpdateLastPassword": false, // Set true to allow PassCore to  update the last password timestamp
    // The following options are for LDAP Provider (remove if you don't use this Provider)
    "LdapSearchBase": "OU=Users,OU=XXXX,DC=XXXX,DC=XXXX",
    "LdapSecureSocketLayer": true, // Default for AD is true when using LDAPS 636
    "LdapStartTls": false, // Default for AD is true when using LDAP 389
    "LdapChangePasswordWithDelAdd": true,
    "LdapSearchFilter": "(sAMAccountName={Username})", // Another value: "(&(objectClass=person)(cn={Username}))"
    // General options (valid for both providers)
    "LdapHostnames": [ "Auth01.XXXX.XXXX" ], // Set your hostname(s)
    "LdapPort": 636, // Default for AD is 389, for LDAPS 636
    "LdapUsername": "CN=Password Reset LDAP,OU=Service Accounts,OU=XXXX,DC=XXXX,DC=XXXX", // Set the username or distinguish name (DN) to bind the LDAP server
    "LdapPassword": "XXXX", // Set the password for the username
    "DefaultDomain": "XXXX.XXXX" // Set your default AD domain here, or non "@" logins will not work! Use empty value to allow user to set the domain. This option is ONLY available with UPN.
  },
  "ClientSettings": {
    "ValidationRegex": {
      "EmailRegex": "^[a-zA-Z0-9.!#$%&’*+/=?^_`{|}~-]+@[a-zA-Z0-9-]+(?:\\.[a-zA-Z0-9-]+)*$",
      "UsernameRegex": "^[a-zA-Z0-9._-]{3,20}$"
    },
    "UsePasswordGeneration": false, //Set true to let PassCore create a new password for the current account. If true the user can not customize its new password.
    "MinimumDistance": 0, //The minimum distance beetween the old and the new password, this is used to enforce the edit distance using the levenshtein distance algorithm.
    "PasswordEntropy": 16, // the number of bytes of entropy to use for generated passwords
    "ShowPasswordMeter": true,
    "MinimumScore": 0, //The minimum acceptable score that the user's new password needs to get at being evaluated by ZXCVBN to be established as the new password.
    "Recaptcha": {
      "SiteKey": "", // ReCAPTCHA public key: replace this! or leave empty if you don't need ReCAPTCHA
      "PrivateKey": "", // ReCAPTCHA private key: replace this! or leave empty if you don't need ReCAPTCHA
      "LanguageCode": "en"
    },
    "UseEmail": "true",
    "ApplicationTitle": "Change Account Password | Self-Service Account Management Tools",
    "ChangePasswordTitle": "Change Account Password",
    "ChangePasswordForm": {
      "HelpText": "If you are having trouble with this tool, please contact IT Support",
      "UsernameLabel": "Username",
      "UsernameHelpblock": "Your organization's email address",
      "UsernameDefaultDomainHelperBlock": "Your organization's username",
      "CurrentPasswordLabel": "Current Password",
      "CurrentPasswordHelpblock": "Enter your current password",
      "NewPasswordLabel": "New Password",
      "NewPasswordHelpblock": "Enter a <a href='https://support.microsoft.com/en-us/help/4026406/microsoft-account-create-a-strong-password' target='_blank'>strong password</a>. You can use <a href='https://xkpasswd.net/s/' target='_blank'>this tool</a> to help you create one; use the XKCD (random sep, pad digit), or NTLM, options.",
      "NewPasswordVerifyLabel": "Re-enter New Password",
      "NewPasswordVerifyHelpblock": "Enter your new password again",
      "ChangePasswordButtonLabel": "Change Password"
    },
    "ErrorsPasswordForm": {
      "FieldRequired": "This field is required",
      "UsernamePattern": "Please enter a valid username",
      "UsernameEmailPattern": "Please enter a valid email address",
      "PasswordMatch": "Passwords do not match"
    },
    "Alerts": {
      "SuccessAlertTitle": "You have changed your password successfully.",
      "SuccessAlertBody": "Please note it may take a few hours for your new password to reach all domain controllers.",
      "ErrorPasswordChangeNotAllowed": "You are not allowed to change your password. Please contact your system administrator.",
      "ErrorInvalidCredentials": "You need to provide the correct current password.",
      "ErrorInvalidDomain": "You have supplied an invalid domain to logon to.",
      "ErrorInvalidUser": "We could not find your user account.",
      "ErrorCaptcha": "Could not verify you are not a robot.",
      "ErrorFieldRequired": "Fulfill all the fields.",
      "ErrorFieldMismatch": "The passwords do not match.",
      "ErrorComplexPassword": "Failed due to password complex policies: New password length is shorter than AD minimum password length",
      "ErrorConnectionLdap": "Unhandled error connecting to the LDAP server.",
      "ErrorScorePassword": "The password you are trying to set is not secure enough.",
      "ErrorDistancePassword": "The password you are trying to set is not diferent enough of your last password.",
      "ErrorPwnedPassword": "The password you are trying to use is publicly known and can be used in dictionary attacks."
    }
  }
}
- Log file (without sensitive information): 

**Describe the bug**
- I have a test user I created in AD (Only membership group is default Domain Users) and I am trying to change the password but I am getting error "ErrorPasswordChangeNotAllowed": "You are not allowed to change your password. Please contact your system administrator."
- My LDAP bind account is a part of the Domain Admins Group (For Testing Purposes)
- I tested my LDAPS (636) connection via built in tool ldp.exe and I was able to successfully bind to the AD server.

**To Reproduce**
1. Go to the web page
2. Put in users information with new password
3. Receive Error "You are not allowed to change your password. Please contact your system administrator."

**Expected behavior**
Test user password gets changed

**Desktop (please complete the following information):**
 - OS: Windows Server 2019
 - Browser: Chrome
 - Version: 85.0.4183.83 (Official Build) (64-bit)

**Additional context**
Verified the following boxes in AD are NOT checked:
- User must change password at next logon
- User cannot change password
- Password never expires
- Account is disabled
Also.... the test account is NOT locked.

**Comment by keagylabs** [comment]
I've got past the issue and am receiving "The server is not operational" now.

**Comment by LingTjal** [comment]
I have the same issue, how did you resolve it?

**Comment by keagylabs** [comment]
I did not resolve it. I feel this current Passcore release is a little buggy. I got past that error code and it started throwing errors such as server is not operational…. When I could bind to LDAPS via ldp.exe

 

I used past releases of Passcore (About 2 Years ago) and it worked out of the box very easy last time. This time I spent way too much time getting it to work.

 

I will possibly try in a future release.
